src/main/java/ch/wisv/areafiftylan/users/controller/UserProfileRestController.java

```diff
@@ -77,22 +78,28 @@
     @PreAuthorize("isAuthenticated()")
     @PostMapping("/current/profile")
     public ResponseEntity<?> addProfile(@AuthenticationPrincipal User user, @Validated @RequestBody ProfileDTO input) {
-        // Check profile for existing rfidLinks
-        boolean isUserCheckedIn = rfidService.isOwnerLinked(user.getEmail());
-
-        if (!isUserCheckedIn) {
-            return this.addProfile(user.getId(), input);
-        }
-
-        LocalDate currentBirthday = user.getProfile().getBirthday();
-        boolean isDateChanged = currentBirthday != input.getBirthday();
-
-        // If rfidLinks are present and the date is changed then return an error
-        if (isDateChanged) {
+        if (checkAllowedToEditProfile(user, input)) {
+            User changedUser = userService.addProfile(user.getId(), input);
+            return createResponseEntity(HttpStatus.OK, "Profile successfully changed", changedUser.getProfile());
+        } else {
             return createResponseEntity(HttpStatus.BAD_REQUEST, "Unable to change date during event", user.getProfile());
```

Comment:
Not permitted maybe? The request itself is valid.

Reply:
Done!